### PR TITLE
setting password auth to true through generator resources

### PR DESCRIPTION
**Pull request by jhoreman**
### Which issue this PR addresses:

This change will allow us to use the VMSS JIT approval in order to access prod and int RPs and GWYs

### What this PR does / why we need it:

This change will allow us to use the VMSS JIT approval in order to access prod and int RPs and GWYs

### Test plan for issue:

https://msazure.visualstudio.com/AzureRedHatOpenShift/_wiki/wikis/ARO.wiki/235246/JIT-VMSS-SSH-Access

### Is there any documentation that needs to be updated for this PR?

https://msazure.visualstudio.com/AzureRedHatOpenShift/_wiki/wikis/ARO.wiki/235246/JIT-VMSS-SSH-Access

### CODE REVIEW

File: pkg/deploy/generator/resources_gateway.go [hunk 1]
```diff
@@ -644,6 +644,9 @@ for scan in baseline clamav software; do
   /usr/local/bin/azsecd config -s $scan -d P1D
 done
 
+# We need to manually set PasswordAuthenticatin to true in order for the VMSS Access JIT to work
```
Comment (s-amann) [inline pkg/deploy/generator/resources_gateway.go:647]: typo
```suggestion
# We need to manually set PasswordAuthentication to true in order for the VMSS Access JIT to work
```

File: pkg/deploy/generator/resources_rp.go [hunk 2]
```diff
@@ -1071,6 +1071,9 @@ for scan in baseline clamav software; do
   /usr/local/bin/azsecd config -s $scan -d P1D
 done
 
+# We need to manually set PasswordAuthenticatin to true in order for the VMSS Access JIT to work
```
Comment (s-amann) [inline pkg/deploy/generator/resources_rp.go:1074]: typo
```suggestion
# We need to manually set PasswordAuthentication to true in order for the VMSS Access JIT to work
```

File: pkg/deploy/generator/resources_gateway.go [hunk 3]
```diff
@@ -644,6 +644,9 @@ for scan in baseline clamav software; do
...
+sed -i 's/PasswordAuthentication no/PasswordAuthentication yes/g' /etc/ssh/sshd_config
```
Comment (s-amann) [inline pkg/deploy/generator/resources_gateway.go:648]: is `PasswordAuthentication no` always in `/etc/ssh/sshd_config`?

**Review by s-amann — APPROVED** [review]
lgtm other than minor typo in comment and a small question

File: pkg/deploy/generator/resources_gateway.go (hunk 3 above)
Comment (jhoreman) [reply pkg/deploy/generator/resources_gateway.go:648]: yes, it actually appears twice. Once commented out and once uncommented and always with the "no" value.

File: pkg/deploy/generator/resources_gateway.go (hunk 1 above)
Comment (jhoreman) [reply pkg/deploy/generator/resources_gateway.go]: fixed, thanks

File: pkg/deploy/generator/resources_rp.go (hunk 2 above)
Comment (jhoreman) [reply pkg/deploy/generator/resources_rp.go]: fixed, thanks

**Review by s-amann — DISMISSED** [review]

**Review by ross-bryan — APPROVED** [review]
lgtm

**Review by bennerv — APPROVED** [review]
lgtm

**Review by s-amann — APPROVED** [review]

**Pull request merged**